app/routes/component-tree.js

```diff
@@ -0,0 +1,47 @@
+import TabRoute from "ember-inspector/routes/tab";
+
+export default TabRoute.extend({
+  setupController() {
+    this._super(...arguments);
+    this.get('port').on('view:viewTree', this, this.setViewTree);
+    this.get('port').on('view:stopInspecting', this, this.stopInspecting);
+    this.get('port').on('view:startInspecting', this, this.startInspecting);
+    this.get('port').on('view:inspectDOMElement', this, this.inspectDOMElement);
+    this.get('port').on('view:inspectComponent', this, this.inspectComponent);
+    this.get('port').send('view:setOptions', { options: this.get('controller.options') });
+    this.get('port').send('view:getTree');
+  },
+
+  deactivate() {
+    this.get('port').off('view:viewTree', this, this.setViewTree);
+    this.get('port').off('view:stopInspecting', this, this.stopInspecting);
+    this.get('port').off('view:startInspecting', this, this.startInspecting);
```

Comment:
Sorry, I wasn't clear on my feedback for this. I mean instead of having a `startInspecting` and `stopInspecting` function, have a single `toggleInspecting` function. Then here, instead of doing `this.startInspecting` you would do `this.toggleInspecting(true)` etc. Just trying to have one function to do starting and stopping, rather than duplicating, but we definitely don't want a third function. Maybe we should just keep the two, and not worry about it.

Reply:
Ha! Easily done.  I like that version much better than the one I wrote